streams/src/main/java/org/apache/kafka/streams/state/ReadOnlyKeyValueStore.java

```diff
@@ -107,4 +108,18 @@
      * @throws InvalidStateStoreException if the store is not initialized
      */
     long approximateNumEntries();
+
+    /**
+     * Get an iterator over keys which have the specified prefix. The type of the prefix can be different from that of
+     * the key. That's why, callers should also pass a serializer for the prefix to convert the prefix into the
+     * format in which the keys are stored underneath in the stores
```

Comment:
We usually put every sentence on its own line In javadocs. Unfortunately, we are not always consistent as you can see in the javadocs in this file.
I guess this iterator has similar requirements as the other iterators, i.e., it must be closed, it must be safe from `java.util.ConcurrentModificationException`, and it must not return `null` values. If so, please add this requirements to the javadocs.
```suggestion
     * Return an iterator over all keys with the specified prefix. 
     * Since the type of the prefix can be different from that of the key, a serializer to convert the
     * prefix into the format in which the keys are stored in the stores needs to be passed to this method.  
```